core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -730,6 +743,11 @@ public static boolean le(String b0, String b1) {
     return b0.compareTo(b1) <= 0;
   }
 
+  /** SQL <code>&le;</code> operator applied to String values. */
```

Comment:
Possibly we could deprecate/remove the old methods (e.g., `le(String b0, String b1)` and always use the new alternatives with the `Comparator`. When we don't have one we could use `String::compareTo` or `Comparator.naturalOrder` to have the same behavior as before.

Reply:
I prefer to leave them for backwards compatibility